src/Sulu/Bundle/AdminBundle/Resources/js/views/FormOverlayList/FormOverlayList.js

```diff
@@ -0,0 +1,199 @@
+// @flow
+import React, {Fragment} from 'react';
+import {observer} from 'mobx-react';
+import {action, observable, toJS} from 'mobx';
+import type {ViewProps} from '../../containers/ViewRenderer';
+import Overlay from '../../components/Overlay';
+import {translate} from '../../utils/Translator';
+import Form from '../../containers/Form';
+import ResourceStore from '../../stores/ResourceStore';
+import List from '../List';
+import ResourceFormStore from '../../containers/Form/stores/ResourceFormStore';
+import formOverlayListStyles from './formOverlayList.scss';
+import ErrorSnackbar from './ErrorSnackbar';
+
+@observer
+export default class FormOverlayList extends React.Component<ViewProps> {
+    static getDerivedRouteAttributes = List.getDerivedRouteAttributes;
+
+    listRef: ?List;
+    formRef: ?Form;
+
+    @observable formStore: ?ResourceFormStore;
+    @observable formErrors = [];
+
+    handleItemAdd = () => {
+        this.createFormOverlay(undefined);
+    };
+
+    handleItemClick = (itemId: string | number) => {
+        this.createFormOverlay(itemId);
+    };
+
+    handleFormOverlayConfirm = () => {
+        if (this.formRef) {
+            this.formRef.submit();
+        }
+    };
+
+    handleFormOverlayClose = () => {
+        this.destroyFormOverlay();
+    };
+
+    handleFormSubmit = () => {
+        if (this.formStore) {
```

Comment:
I would rewrite that part. I would throw an error at the very beginning if the formStore is not defined, telling that the form store is missing, and that this is likely a bug. Have similar errors in other places.

Because it should not happen that the formstore is not defined, right? Could lead to confusing situations otherwise...